tools/ThermoRawFileParser/thermo_converter.xml

```diff
@@ -1,60 +1,193 @@
     <description>RAW file converter</description>
     <requirements>
-        <requirement type="package" version="1.1.2">ThermoRawFileParser</requirement>
+        <requirement type="package" version="1.1.10">ThermoRawFileParser</requirement>
     </requirements>
     <command>
 <![CDATA[
+#import re
 
-ln -s '$input' ./input.raw &&
+#set $temp_stderr = "thermo_converter_stderr"
 
-ThermoRawFileParser.sh
-    -i=input.raw
-    -o=./
+echo "" > $temp_stderr &&
+
+mkdir ./raws_folder &&
+mkdir ./output_folder &&
+#for $input_raw in $input:
+    #if len($input) > 1
+        #set $input_name = re.sub('[^\w\-\.]', '_',$input_raw.element_identifier.split('/')[-1].replace(".raw", "") + ".raw")
+        ln -s -f '${input_raw}' './raws_folder/${input_name}' &&
+    #else:
+        ln -s -f '${input_raw}' './raws_folder/input.raw' &&
+    #end if
+#end for
+
+(ThermoRawFileParser.sh
+    -d=./raws_folder
+    -o=./output_folder
     -f=$output_format
-    #if $metadata:
-        --metadata=1
+    #if $output_metadata_selector != "off":
+        --metadata="${output_metadata_selector}"
     #end if
+    $zlib_boolean
+    $peakpicking_boolean
+    $ignore_instrument_errors_boolean
+
+    2>> $temp_stderr)
+
     &&
-    #if $output_format == '0':
-        mv ./input.mgf '$output'
-    #else:
-        mv ./input.mzML '$output'
+
+    #if len($input) == 1:
+        #if $output_format == "0":
+            mv ./output_folder/input.mgf ./output_file.out &&
+        #else if $output_format == "1":
+            mv ./output_folder/input.mzML ./output_file.out &&
+        #else if $output_format == "2":
+            mv ./output_folder/input.mzML ./output_file.out &&
+        #end if
+
+        #if $output_metadata_selector != "off":
+            #if $output_metadata_selector == "0":
+                mv ./output_folder/input-metadata.json ./input-metadata.txt &&
+            #else if $output_metadata_selector == "1":
+                mv ./output_folder/input-metadata.txt ./input-metadata.txt &&
+            #end if
+        #end if
     #end if
+
+    cat $temp_stderr 2>&1;
 ]]>
     </command>
     <inputs>
-        <param name="input" type="data" format="thermo.raw" label="Thermo RAW file" help="" />
-        <param name="metadata" type="boolean" truevalue="" falsevalue="" checked="False"
-            label="Output metadata" help="" />
-        <param name="output_format" type="select" label="Choose an output format">
+        <param name="input" type="data" format="thermo.raw" label="Thermo RAW file" help="" multiple="true"
+          optional="False" />
+
+        <param name="output_format" type="select" label="Output format">
             <option value="0">mgf</option>
             <option value="1" selected="True">mzml</option>
+            <option value="2">Indexed mzml</option>
         </param>
+
+        <param name="zlib_boolean" type="boolean" truevalue="" falsevalue="-z" checked="true"
```

Comment:
Is this parameter still needed if we do not offer compressed outputs?

Reply:
Well, zlib compression is used by default (-z flag disables it, not the opposite) and its compression is quite notorious (4mb size vs 2mb, applied to the small small.raw file - 1.4 mb ) although. I am not an expert but I think the ability of disabling this compression may be useful in some contexts without proper zlib management or for readability purposes...